x-pack/plugins/fleet/server/routes/agent/handlers.ts

```diff
@@ -323,13 +323,22 @@ export const getAgentStatusForAgentPolicyHandler: FleetRequestHandler<
     const [coreContext, fleetContext] = await Promise.all([context.core, context.fleet]);
     const esClient = coreContext.elasticsearch.client.asInternalUser;
     const soClient = fleetContext.internalSoClient;
+
+    const parsePolicyIds = (policyIds: string | string[] | undefined): string[] | undefined => {
```

Comment:
Could you add some tests to cover these changes? It would be good especially to test the behaviour of `getAgentStatusForAgentPolicy` in the different cases. Thanks!